wdae/wdae/users_api/views.py

```diff
@@ -543,10 +541,43 @@ def delete(self, request):
         user = request.user
         if not user.is_authenticated:
             return Response(status=status.HTTP_401_UNAUTHORIZED)
+        if not get_application_model() \
+                .objects \
+                .filter(name=request.data.get("name")) \
+                .exists():
+            return Response(status=status.HTTP_400_BAD_REQUEST)
         app = get_application_model().objects.get(
             name=request.data.get("name")
         )
         if not user.id == app.user_id:
             return Response(status=status.HTTP_401_UNAUTHORIZED)
         app.delete()
         return Response(status=status.HTTP_200_OK)
+
+    @request_logging(LOGGER)
+    def put(self, request):
+        """Update a given federation token's name."""
+        user = request.user
+        if not user.is_authenticated:
+            return Response(status=status.HTTP_401_UNAUTHORIZED)
+        if not get_application_model() \
+                .objects \
+                .filter(name=request.data.get("name")) \
+                .exists():
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        if (get_application_model()
+                .objects
+                .filter(name=request.data.get("new_name"))
+                .exists()):
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        app = get_application_model().objects.get(
+            name=request.data.get("name")
+        )
+        if not user.id == app.user_id:
+            return Response(status=status.HTTP_401_UNAUTHORIZED)
+        app.name = request.data.get("new_name")
```

Comment:
You should add a check that `new_name` is both present in the request data and has a value (is not None). If this check fails, return a 400 response.

Reply:
After that you could convert this `.get` access into bracket notation (i.e. `request.data["new_name"]`) - this clearly conveys that we expect `new_name` to be present in the dict since this will throw a `KeyError` if it is not found (in contrast to `.get`, which will just return a default value of `None`)